server/lib/oauth2/oauth2.executeQuery.js

```diff
@@ -0,0 +1,136 @@
+  const currentProvider = providers[integrationName];
+  const { tokenHost } = currentProvider;
+  const { tokenPath } = currentProvider;
+  const { authorizeHost } = currentProvider;
+  const { authorizePath } = currentProvider;
+  const { integrationScope } = currentProvider;
+
+  // Init credentials based on integration name
+  const credentials = {
+    client: {
+      id: `${clientId}`,
+      secret: `${secretId}`,
+    },
+    auth: {
+      tokenHost,
+      tokenPath,
+      authorizeHost,
+      authorizePath,
+    },
+    http: {
+      json: true,
+    },
+  };
+
+  try {
+    const client = new ClientCredentials(credentials);
+    let authResult = await client.createToken(accessToken);
+
+    if (authResult.expired()) {
+      logger.trace('Refresh token is requiered');
+      // Refresh token
+      try {
+        const refreshParams = {
+          client_id: clientId,
+          client_secret: secretId,
+          scope: integrationScope,
+        };
+
+        authResult = await authResult.refresh(refreshParams);
+
+        // Save new  accessToken
+        await gladys.variable.setValue(
+          `${integrationName.toUpperCase()}_ACCESS_TOKEN`,
+          JSON.stringify(authResult),
+          serviceId,
+          userId,
+        );
+      } catch (error) {
+        logger.error('Error refreshing access token: ', error);
+      }
+    }
+
+    logger.trace('authResult: ', authResult);
+    return authResult;
+  } catch (error) {
+    logger.error(error);
+    return null;
+  }
+}
+
+/**
+ * @description Execute a query to Oauth2 API.
+ * @param {string} serviceId - Gladys service id call method.
+ * @param {string} userId - Gladys userId of current session.
+ * @param {string} integrationName - Oauth2 integration name.
+ * @param {string} queryType - Method of call ('get' or 'post').
+ * @param {string} queryUrl - Url to send query.
+ * @param {string} queryParams - Array of query params.
+ * @returns {Promise} Result of query .
+ * @example
+ * oauth2.executeQuery('7fdsf4s68r4gfr68f4r63csd7f6f4c3r85', '78v4f3df83g74v1fsd8375f63gvrf5c', 'withings'
+ *  'Bearer', 'get', 'http://localhost/apiname',[...]);
+ */
+async function executeQuery(serviceId, userId, integrationName, queryType, queryUrl, queryParams) {
+  // Refresh token access if needed
+  const accesToken = await refreshTokenAccess(serviceId, integrationName, this.gladys, userId);
+
+  const headerConfig = {
+    headers: {
+      Authorization: `${accesToken.token.token_type} ${accesToken.token.access_token}`,
+    },
+  };
+
+  try {
+    let result;
+    if (queryType === 'get' || queryType === 'GET' || queryType === 'Get') {
+      // Add param in query
+      result = await axios.get(`${queryUrl}?${queryParams}`, headerConfig);
+    } else {
+      // Add param to body
+      const data = {};
+      const queryParamsExtract = queryString.parse(queryParams);
+      Object.keys(queryParamsExtract).forEach(function parseQueryParams(key) {
```

Comment:
Could it be replaced by an arrow function?

Reply:
i transform to arrow function